### PR TITLE
tuhi-course-poster-vuw:0.1.0

### PR DESCRIPTION
**Pull request by baptiste**
I am submitting
- [x] a new package
- [ ] an update for a package

This package defines a poster template used to showcase the content of undergraduate courses at Victoria Uni of Wellington (VUW). Logos etc. have been replaced with dummy content in this public template, however.

I have read and followed the submission guidelines and, in particular, I
- [x] selected a name that isn't the most obvious or canonical name for what the package does
- [x] added a `typst.toml` file with all required keys
- [x] added a `README.md` with documentation for my package
- [x] have chosen a license and added a `LICENSE` file or linked one in my `README.md`
- [x] tested my package locally on my system and it worked
- [x] `exclude`d PDFs or README images, if any, but not the LICENSE

- [x] ensured that my package is licensed such that users can use and distribute the contents of its template directory without restriction, after modifying them through normal use.

### CODE REVIEW

File: packages/preview/course-poster-vuw/0.1.0/course-poster-vuw.typ [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+place(
+  dx: margin-body-left,
+  dy: image-top,
+  rect(width:image-width, height:0.5*image-width, stroke: 0.5*stoke-width)
+)
+
+place(
+  dx: margin-body-left,
+  dy:  image-top,
+  image(courseimage, width:image-width, height:0.5*image-width),
```
Comment (laurmaedje) [inline packages/preview/course-poster-vuw/0.1.0/course-poster-vuw.typ:221]: The path will be resolved relative to the package, so this currently breaks after template initialization. To make this user-configurable, you'd need to move the whole image call into the user code. You can still set the image width and height here using a set rule.  You can take a look at the `dashing-dept-news` template to see how that works.

File: packages/preview/course-poster-vuw/0.1.0/typst.toml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+[package]
+name = "course-poster-vuw"
+version = "0.1.0"
+entrypoint = "course-poster-vuw.typ"
+authors = ["baptiste <https://github.com/baptiste>"]
+license = "MPL-2.0"
+description = "A poster template for VUW course descriptions."
+categories = ["office"]
```
Comment (laurmaedje) [inline packages/preview/course-poster-vuw/0.1.0/typst.toml:8]: there is a "poster" category, which is a better fit :)

**Review by laurmaedje — COMMENTED** [review]
Thanks for the template submission! It looks really nice.

As-is the template's name is too descriptive. We have a rule that template names must include some creative/unique component so that multiple people can create a template for the same purpose or institution without one having an unfair naming advantage. See the [submission guidelines](https://github.com/typst/packages?tab=readme-ov-file#submission-guidelines) for details. You can also take a look [at the existing templates](https://typst.app/universe/search?kind=templates) for inspiration.

File: packages/preview/course-poster-vuw/0.1.0/course-poster-vuw.typ (hunk 1 above)
Comment (baptiste) [reply packages/preview/course-poster-vuw/0.1.0/course-poster-vuw.typ]: Thanks, I've changed this.

File: packages/preview/tuhi-course-poster-vuw/0.1.0/template/main.typ [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+courselecturers : ("Bridges", "Stairway",),
+courseformat : text[*2* bridges/hangouts],
+coursedescription : text[Stairway to Heaven _that won't let you down_],
+courseprereqs: text[*1* achievement backpack \
+  in *LEVEL 3 NCEA* Tramping\
+  or equivalent, or *murph101*],
+  courses: courses,
+contact: contact,
+logo: logo,
+qrcodeurl: url,
```
Comment (laurmaedje) [inline packages/preview/tuhi-course-poster-vuw/0.1.0/template/main.typ:49]: You might want to add indent here to improve readability.

File: packages/preview/tuhi-course-poster-vuw/0.1.0/tuhi-course-poster-vuw.pdf
Comment (laurmaedje) [inline packages/preview/tuhi-course-poster-vuw/0.1.0/tuhi-course-poster-vuw.pdf:1]: This file should probably be deleted.

File: packages/preview/tuhi-course-poster-vuw/0.1.0/template/main.typ [hunk 4]
```diff
@@ -0,0 +1,66 @@
+#import "@preview/course-poster-vuw:0.1.0": course-poster-vuw
```
Comment (laurmaedje) [inline packages/preview/tuhi-course-poster-vuw/0.1.0/template/main.typ:1]:
```suggestion
#import "@preview/tuhi-course-poster-vuw:0.1.0": tuhi-course-poster-vuw
```

File: packages/preview/tuhi-course-poster-vuw/0.1.0/template/main.typ [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  Y2T2: "esca241 · esca242 · trek201",
+  Y3T1: "esca305 · esca307 · trek301",
+  Y3T2: "esca304 · esca345 · trek360",
+  Y4T1: "esca411 · esca413 · trek415 · swim490",
+  Y4T2: "esca412 · esca414 · trek416 · swim417",
+)
+
+// use the template
+
+#show: course-poster-vuw.with(
```
Comment (laurmaedje) [inline packages/preview/tuhi-course-poster-vuw/0.1.0/template/main.typ:32]:
```suggestion
#show: tuhi-course-poster-vuw.with(
```

File: packages/preview/tuhi-course-poster-vuw/0.1.0/template/main.typ [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+coursetitle : text[escarpment track],
+courseid : "esca101",
+coursemajor: "hiking",
+courseimage: "images/esca101.jpg",
```
Comment (laurmaedje) [inline packages/preview/tuhi-course-poster-vuw/0.1.0/template/main.typ:36]:
```suggestion
courseimage: image("images/esca101.jpg"),
```

File: packages/preview/tuhi-course-poster-vuw/0.1.0/tuhi-course-poster-vuw.pdf
Comment (baptiste) [reply packages/preview/tuhi-course-poster-vuw/0.1.0/tuhi-course-poster-vuw.pdf:1]: Oops, sorry – I believe it's auto-generated by VS code every time I edit a typ file :/

Comment (laurmaedje) [reply packages/preview/tuhi-course-poster-vuw/0.1.0/tuhi-course-poster-vuw.pdf:1]: Yeah, that's a feature of the (unofficial) LSP, which is somewhat unfortunate in my opinion. I think you can disable it in the settings.

**Comment by laurmaedje** [comment]
Thank you!

**Pull request merged**